engine/src/main/java/io/zeebe/engine/state/appliers/ProcessInstanceElementActivatingApplier.java

```diff
@@ -134,19 +130,14 @@ private void cleanupSequenceFlowsTaken(final ProcessInstanceRecord value) {
         processState.getFlowElement(
             value.getProcessDefinitionKey(), value.getElementIdBuffer(), ExecutableFlowNode.class);
 
-    final var tokensBySequenceFlow =
-        sequenceFlowAnalyzer.determineTakenIncomingFlows(value.getFlowScopeKey(), parallelGateway);
-
-    // cleanup deferred records for sequence flow taken to this parallel gateway
-    tokensBySequenceFlow.forEach(
-        (sequenceFlow, tokens) -> {
-          // before a parallel gateway is activated, each incoming sequence flow of the gateway must
-          // be taken (at least once). if a sequence flow is taken more than once then the redundant
-          // token remains for the next activation of the gateway (Tetris principle)
-          final var firstToken = tokens.get(0);
-          elementInstanceState.removeStoredRecord(
-              value.getFlowScopeKey(), firstToken.getKey(), Purpose.DEFERRED);
-        });
+    if (parallelGateway.getIncoming().size() > 1) {
```

Comment:
what about parallel gateway with `incoming.size() == 1`?

Reply:
It is working with one flow because it is checked in all places. However, I'll remove the check :+1: